src/routes/api/product.route.js

```diff
@@ -15,5 +15,16 @@ route.post(
   validateProduct,
   ProductController.addItem
 );
-
+route.get(
+  "/",
+  extractToken,
+  checkRole(["admin", "seller", "buyer"]),
```

Comment:
If this endpoint is public (if anyone can use it) there is no need to check the role (remove the middleware.